### PR TITLE
python venv

### PR DESCRIPTION
**Pull request by alexandrecuer**
Tested on an emonpi with an emontx v3, works fine. As my usage of emonhub is limited to the basic interfacers (EmonHubJeeInterfacer, EmonHubMqttInterfacer) should be tested more deeply

cf https://github.com/openenergymonitor/EmonScripts/issues/171

to install that version on a new setup, using the scripted installations, just modify the install/config.ini so that it uses my modified branch : `emonhub_branch=master` and `git_repo[emonhub]=https://github.com/alexandrecuer/emonhub.git`

### CODE REVIEW

**Comment by borpin** [comment]
Does the `systemd` service file need adjusting?

I am guessing not but checking.

Would this be a good time to switch to a requirements file so we can specify the versions of the python packages?


https://github.com/openenergymonitor/emonhub/blob/4539b9e928e3f7b5bc3dd1922440cf7e725ac6b9/service/emonhub.service#L10

**Comment by borpin** [comment]
The other issue is migration.

We can't update `emonhub.py` with the different hashbang without installing the venv. So a migration script would be needed.

Remember older versions as well.

**Comment by alexandrecuer** [comment]
No the service file does not need any modification.

For older versions, yes it is a breaking change but is possible to help migrations, without over complicating things ? A possibility could be to change the shebang from the install script. it would be tracked by git as a modification but it is maybe not a big problem....

File: install.sh [hunk 1]
```diff
@@ -37,7 +37,8 @@ else
   echo "user provided as arg = $user"
 fi
 
-sudo apt-get install -y  bluetooth libbluetooth-dev
+sudo apt-get install -y bluetooth libbluetooth-dev
+sudo apt-get install -y python3-venv
 
 sudo mkdir -p /opt/v
 sudo chown $(id -u -n):$(id -u -n) /opt/v
```
Comment (borpin) [inline install.sh:44]: Should this use the `user` set earlier?

File: install.sh [hunk 2]
```diff
@@ -37,7 +37,8 @@ else
   echo "user provided as arg = $user"
 fi
 
-sudo apt-get install -y  bluetooth libbluetooth-dev
+sudo apt-get install -y bluetooth libbluetooth-dev
+sudo apt-get install -y python3-venv
 
 sudo mkdir -p /opt/v
```
Comment (borpin) [inline install.sh:43]: Would it make sense to place this under the emonhub `script_dir` to keep it all together?

File: install.sh (hunk 1 above)
Comment (alexandrecuer) [reply install.sh:44]: good catch

File: install.sh (hunk 2 above)
Comment (alexandrecuer) [reply install.sh:43]: dont understand what you mean

Comment (borpin) [reply install.sh:43]: You are putting the venv directory at what I suggest is a random location. Put it under the variable 'emon_script' location so it keeps everything together.

Comment (alexandrecuer) [reply install.sh:43]: by random location, you mean `/opt/v` ?
maybe yes another name than v could be better, something like `/opt/venv` so the virtual env for emonhub would be in `/opt/venv/emonhub` and not in `/opt/v/emonhub`
maybe you mean `/opt/openenergymonitor/venv` ?

**Comment by alexandrecuer** [comment]
not so usefull after all

**Pull request closed without merging**